src/coreclr/src/jit/rangecheck.cpp

```diff
@@ -59,13 +59,29 @@ int RangeCheck::GetArrLength(ValueNum vn)
     return m_pCompiler->vnStore->GetNewArrSize(arrRefVN);
 }
 
-// Check if the computed range is within bounds.
-bool RangeCheck::BetweenBounds(Range& range, int lower, GenTree* upper)
+//------------------------------------------------------------------------
+// BetweenBounds: Check if the computed range is within bounds
+//
+// Arguments:
+//    Range - the range to check if in bounds
+//    upper - the array length vn
+//    arrSize - the length of the array if known, or <= 0
+//
+// Return Value:
+//    True iff range is between [0 and vn - 1] or [0, arrSize - 1]
+//
+// notes:
+//    This function assumes that the lower range is resolved and upper range is symbolic as in an
+//    increasing loop.
+//
+// TODO-CQ: This is not general enough.
+//
+bool RangeCheck::BetweenBounds(Range& range, GenTree* upper, int arrSize)
```

Comment:
Thanks for this change: 'upper' is the correct name for this parameter, not 'lower'